File: benchmark/perl/fib_iterative_approximation.pl
use Time::HiRes qw(time);
sub fib {
    my ($n) = @_;
    return $n + 0.0 if $n <= 1;
    my $a = 0.0; my $b = 1.0;
    for my $i (2..$n) {
        my $temp = $a + $b;
        $a = $b;
        $b = $temp;
    }
    return $b;
}
my $start = time * 1000;
my $s = 0.0;
for (1..10000) { $s += fib(1000); }
my $end = time * 1000;
print "Sum: $s
";
printf "Time: %.0fms
", $end - $start;

File: benchmark/lua/fib_iterative_approximation.lua
-- Note: Lua uses float for large numbers
local start = os.clock()
local function fib(n)
    if n <= 1 then return n + 0.0 end
    local a, b = 0.0, 1.0
    for i = 2, n do
        local temp = a + b
        a = b
        b = temp
    end
    return b
end
local s = 0.0
for i = 1, 10000 do s = s + fib(1000) end
local finish = os.clock()
print("Sum: " .. s)
print(string.format("Time: %.0fms", (finish - start) * 1000))